packages/bson-transpilers/symbols/php/templates.yaml

```diff
@@ -0,0 +1,1271 @@
+        (lhs, arg) => {
+            return ` ${arg}`;
+        }
+    LongGreaterThanOrEqualTemplate: &LongGreaterThanOrEqualTemplate !!js/function >
+        (lhs) => {
+            return `${lhs} >=`;
+        }
+    LongGreaterThanOrEqualArgsTemplate: &LongGreaterThanOrEqualArgsTemplate !!js/function >
+        (lhs, arg) => {
+            return ` ${arg}`;
+        }
+    LongLessThanTemplate: &LongLessThanTemplate !!js/function >
+        (lhs) => {
+            return `${lhs} <`;
+        }
+    LongLessThanArgsTemplate: &LongLessThanArgsTemplate !!js/function >
+        (lhs, arg) => {
+            return ` ${arg}`;
+        }
+    LongLessThanOrEqualTemplate: &LongLessThanOrEqualTemplate !!js/function >
+        (lhs) => {
+            return `${lhs} <=`;
+        }
+    LongLessThanOrEqualArgsTemplate: &LongLessThanOrEqualArgsTemplate !!js/function >
+        (lhs, arg) => {
+            return ` ${arg}`;
+        }
+    LongFloatApproxTemplate: &LongFloatApproxTemplate !!js/function >
+        (lhs) => {
+            return `(float) ${lhs}`;
+        }
+    LongTopTemplate: &LongTopTemplate !!js/function >
+        (lhs) => {
+            return `${lhs} >> 32`;
+        }
+    LongBottomTemplate: &LongBottomTemplate !!js/function >
+        (lhs) => {
+            return `${lhs} & 0x0000ffff`;
+        }
+    TimestampToStringTemplate: &TimestampToStringTemplate !!js/function >
+        (lhs) => {
+            return `(string) ${lhs}`;
+        }
+    TimestampToStringArgsTemplate: &TimestampToStringArgsTemplate !!js/function >
+        () => {
+            return '';
+        }
+    TimestampEqualsTemplate: &TimestampEqualsTemplate !!js/function >
+        (lhs) => {
+            return `${lhs} == `;
+        }
+    TimestampEqualsArgsTemplate: &TimestampEqualsArgsTemplate !!js/function >
+        (lhs, arg) => {
+            return `${arg}`;
+        }
+    TimestampGetLowBitsTemplate: &TimestampGetLowBitsTemplate !!js/function >
+        (lhs) => {
+            return `(${lhs})->getTimestamp()`;
+        }
+    TimestampGetLowBitsArgsTemplate: &TimestampGetLowBitsArgsTemplate !!js/function >
+        () => {
+            return ''
+        }
+    TimestampGetHighBitsTemplate: &TimestampGetHighBitsTemplate !!js/function >
+        (lhs) => {
+            return `(${lhs})->getIncrement()`;
+        }
+    TimestampGetHighBitsArgsTemplate: &TimestampGetHighBitsArgsTemplate !!js/function >
+        () => {
+            return ''
+        }
+    TimestampTTemplate: &TimestampTTemplate !!js/function >
+        (lhs) => {
+            return `(${lhs})->getTimestamp()`;
+        }
+    TimestampITemplate: &TimestampITemplate !!js/function >
+        (lhs) => {
+            return `(${lhs})->getIncrement()`;
+        }
+    TimestampAsDateTemplate: &TimestampAsDateTemplate !!js/function >
+        (lhs) => {
+            return `(new \\DateTime('1970-1-1', new \\DateTimeZone('UTC')))->setTimestamp((${lhs})->getTimestamp())`;
```

Comment:
Since the Timestamp counts from the epoch, I think the simplest code would be to use `new DateTime('@<timestamp>`). The timezone would implicitly be UTC for that construction.

Alternatively, if you need to return a UTCDateTime, you can just use [its constructor](https://www.php.net/manual/en/mongodb-bson-utcdatetime.construct.php) and multiply the timestamp by 1000 to obtain milliseconds since the epoch (bypassing DateTime entirely).

Reply:
This template was completely changed with UTCDateTime implementation.

Reply:
OK to resolve.